### PR TITLE
Feature: Looker role data source

### PR DESCRIPTION
**Pull request by jessicasomaiya**
## Description

This PR implements a `data_source` that allows the provider user to retrieve a `looker_role` given a role name. The attributes that are accessible from this data source are `id` `name`, `model_set` and `permission_set` . `model_set` and `permission_set` have nested attributes such as `id' and 'name'. 

I took inspiration from [this documentation](https://learn.hashicorp.com/tutorials/terraform/provider-complex-read#implement-complex-read). 

## What is the need for a looker_role data source? 

The allocate a role such as `admin` to a user, you need the `looker_role.id`. This data source allows the user of the provider to retrieve role ids for roles that already exist, such as those created by looker for free. For example, `developer` and `viewer`.

Note that the `developer`, `viewer` and `user` roles created by looker _can_ be changed (on the UI and via the API). The `admin` looker role cannot be changed.  If a user wanted to change the permissions/models of an existing role, this is not currently supported by the terraform provider - if there is a need this can be considered as an additional feature.

## Has this been tested?

Yes! This implementation has been tested locally. Unit tests are on the todo list!

### CODE REVIEW

File: docs/data-sources/role.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+# looker_role (Data Source)
+
+## Example Usage
+```terraform
+data "looker_role" "developer" {
+  name = "Developer"
+}
+
+output "developer" {
```
Comment (ponteilla) [inline docs/data-sources/role.md:17]:
```suggestion
output "developer_model_set_models" {
```

File: examples/data-sources/role/main.tf [hunk 2]
```diff
@@ -0,0 +1,7 @@
+data "looker_role" "developer" {
+  name = "Developer"
+}
+
+output "developer" {
```
Comment (ponteilla) [inline examples/data-sources/role/main.tf:5]:
```suggestion
output "developer_model_set_models" {
```

File: looker/data_source_looker_role.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+func datasourceRoleRead(ctx context.Context, d *schema.ResourceData, c interface{}) diag.Diagnostics {
+	api := c.(*sdk.LookerSDK)
+
+	// exactly one of these variables will be nil - this is enforced by the data source schema
+	name := conv.PString(d.Get("name").(string))
+	id := conv.PString(d.Get("id").(string))
+
+	roles, roleErr := api.SearchRoles(sdk.RequestSearchRoles{}, nil)
```
Comment (ponteilla) [inline looker/data_source_looker_role.go:99]: Should we have `Id` and `Name` on the request struct?

File: looker/data_source_looker_role.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+	)
+
+	return diag.FromErr(result.ErrorOrNil())
+}
+
+// flattenModelSet takes a model set object and maps the field from the object to a map[string]interface{}. The key is the attribute name defined in the data source
+// schema, and the value is the corresponding value coming from the looker api. This function returns a []interface{} because the
+// terraform aggregate type schema.TypeSet expects a slice of attributes.
+func flattenModelSet(m *sdk.ModelSet) []interface{} {
+	ms := make(map[string]interface{}, 0)
```
Comment (ponteilla) [inline looker/data_source_looker_role.go:134]: Is it possible to have a role with a no permission or model set?

```suggestion
func flattenModelSet(m *sdk.ModelSet) []interface{} {
  if m == nil {
    return make([]interface{}, 0)
  }
  
	ms := make(map[string]interface{}, 0)
```

File: looker/data_source_looker_role.go (hunk 3 above)
Comment (jessicasomaiya) [reply looker/data_source_looker_role.go:99]: Yes! Definitely - thank you :)

File: looker/data_source_looker_role.go (hunk 4 above)
Comment (jessicasomaiya) [reply looker/data_source_looker_role.go:134]: Yes it is - in which case this would be better :)

Comment (jessicasomaiya) [reply looker/data_source_looker_role.go:134]: Actually I just tried creating a role using the API without a permission set or model set and it's not possible.

**Review by ponteilla — APPROVED** [review]

**Pull request merged**